code/HabitTracker/app/src/main/java/com/example/habittracker/utils/DateConverter.java

```diff
@@ -33,24 +34,22 @@
                         day
                 )));
     }
-
-    /**
-     * converts date array list to string
-     * @param date_list
-     * @return
-     */
+  
     @RequiresApi(api = Build.VERSION_CODES.N)
     public static String arrayListToString (ArrayList<Integer> date_list) {
-        String date = date_list.stream().map(String::valueOf)
-                .collect(Collectors.joining("-"));
+
+        String date = "";
+        for (int i = 0;i<date_list.size();i++) {
+            if(i<date_list.size()-1) {
+                date = date + date_list.get(i) + "-";
+            }
+            else {
+                date = date + date_list.get(i);
+            }
+        }
         return date;
     }
 
-    /**
-     * convert string to arraylist
-     * @param date_string
-     * @return
-     */
     @RequiresApi(api = Build.VERSION_CODES.N)
     public static ArrayList<Integer> stringToArraylist (String date_string) {
         String [] temp = date_string.split("-");
```

Comment:
Sorry I can't comment in it. But it seems like you have also made an arraylistToDate function.

Which I also have. we can keep yours but I see some possible problems arising because I had to make mine take an ArrayList of Long. Otherwise it would not work. I will drop it here so you can decide what to keep/trash.

```
    /**
     * Converts a given arraylist to a Date
     * @param dateArrayList {ArrayList<Long>}   given in the following format: {year, month, day}
     * @return  date, the converted Date
     */
    public static Date arrayListToDate (ArrayList<Long> dateArrayList) {
        Date date = null;

        if (dateArrayList != null) {
            long year = dateArrayList.get(0);
            long month = dateArrayList.get(1);
            long day = dateArrayList.get(2);
            String dateAsString = String.format(Locale.US, "%d/%d/%d", month, day, year);

            try {
                date = new SimpleDateFormat("MM/dd/yyyy", Locale.US).parse(dateAsString);
            } catch (ParseException e) {
                e.printStackTrace();
            }
        }
        return (date);
    }
```

Reply:
I use this method in my progress section, I believe the implementation in the dev branch works and it uses a list of Integers instead of long. There is something weird going on with Boxing of the Integer type which I don't fully understand, but I figured something out I guess

Reply:
I'm cool with this then, I'll have to playa round with it when it's my time to merge.